scalyr_agent/agent_main.py

```diff
@@ -1658,6 +1749,59 @@ def __calculate_overall_stats(self, base_overall_stats):
             result.rss_size,
         ) = self.__controller.get_usage_info()
 
+        if copy_manager_warnings:
+            result.avg_bytes_copied_rate = (
+                result.total_bytes_copied - self.__last_total_bytes_copied
+            ) / self.__config.copying_manager_stats_log_interval
+            result.avg_bytes_produced_rate = (
+                (
+                    result.total_bytes_skipped
+                    + result.total_bytes_copied
+                    + result.total_bytes_pending
+                )
+                - (
+                    self.__last_total_bytes_skipped
+                    + self.__last_total_bytes_copied
+                    + self.__last_total_bytes_pending
+                )
+            ) / self.__config.copying_manager_stats_log_interval
+            if result.total_bytes_skipped > self.__last_total_bytes_skipped:
+                rate_limited_time = (
+                    result.total_rate_limited_time - self.__last_total_rate_limited_time
+                ) / self.__config.copying_manager_stats_log_interval
+                if self.__config.parsed_max_send_rate_enforcement:
```

Comment:
Let's add tests for those two warnings.

Will likely need to mock a couple of things, but it should be mostly straight forward if we test this method directly.

Reply:
And eventually (post release is fine) we should also have some end tests which would cover that in combination with the new ``max_send_rate_enforcement`` configuration option.

We now have end to end tests in place so it shouldn't be too painful to add a test case for that scenario.